### PR TITLE
feat: distinction between class and instance variables

### LINKED ISSUE #129: Distinction between class and instance variables
### Is your feature request related to a problem?

Currently, there is no distinction between class variables and instance variables when analysing the scope of a class.

### Desired solution

Add a function to distinguish between class and instance variables. It should return two lists.

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by lukarade**
Closes #129 

### Summary of Changes

The `get_scope` function now has two additional fields: `instance_variables` and `class_variables`. Each contains the corresponding variables for a class node.

### CODE REVIEW

File: src/library_analyzer/processing/api/_resolve_references.py [hunk 1]
```diff
@@ -120,26 +177,42 @@ def enter_assignname(self, node: astroid.AssignName) -> None:
...
             parent = self.current_node_stack[-1]
-            scope_node = ScopeNode(node=node, children=None, parent=parent)
+            scope_node = ScopeNode(node=node, children=[], parent=parent)
             self.children.append(scope_node)
 
-    def enter_assignattr(self, node: astroid.Attribute) -> None:
+        # add class variables to the class_variables list of the class
+        if isinstance(node.parent.parent, astroid.ClassDef):
+            class_node = self.get_node_by_name(node.parent.parent.name)
+            if class_node is not None and isinstance(class_node, ClassScopeNode):
```
Comment (nvollroth) [inline src/library_analyzer/processing/api/_resolve_references.py:186]: see line 120

File: src/library_analyzer/processing/api/_resolve_references.py [hunk 2]
```diff
@@ -120,26 +177,42 @@ def enter_assignname(self, node: astroid.AssignName) -> None:
...
         parent = self.current_node_stack[-1]
-        scope_node = ScopeNode(node=node, children=None, parent=parent)
+        scope_node = ScopeNode(node=node, children=[], parent=parent)
         self.children.append(scope_node)
 
     def enter_importfrom(self, node: astroid.ImportFrom) -> None:
         parent = self.current_node_stack[-1]
-        scope_node = ScopeNode(node=node, children=None, parent=parent)
+        scope_node = ScopeNode(node=node, children=[], parent=parent)
         self.children.append(scope_node)
```
Comment (nvollroth) [inline src/library_analyzer/processing/api/_resolve_references.py:203]: Does it really make sense to create two independent functions here that differ only in name and in a parameter type, but do exactly the same thing?

I would combine the two functions into one and restrict the parameter node to the types astroid.Import and astroid.FromImport, as you are creating redundancy here.

File: src/library_analyzer/processing/api/_resolve_references.py [hunk 3]
```diff
@@ -83,25 +133,32 @@ def enter_module(self, node: astroid.Module) -> None:
...
+                node=node,
+                children=[],
+                parent=self.current_node_stack[-1],
+                instance_variables=[],
+                class_variables=[],
+            ),
         )
 
     def leave_classdef(self, node: astroid.ClassDef) -> None:
         self.detect_scope(node)
```
Comment (nvollroth) [inline src/library_analyzer/processing/api/_resolve_references.py:154]: Does it really make sense to create two independent functions(l.139-140, l.153)  here that differ only in name and in a parameter type, but do exactly the same thing?

I would combine the two functions into one and restrict the parameter node to the types astroid.Module and astroid.ClassDef, as you are creating redundancy here.

**Review by nvollroth — COMMENTED** [review]
On the whole, the whole thing looks very good.

File: src/library_analyzer/processing/api/_resolve_references.py (hunk 2 above)
Comment (lukarade) [reply src/library_analyzer/processing/api/_resolve_references.py:203]: The separate functions are absolutely necessary. 
The AST is traversed by the ASTWalker, which needs a separate enter and leave function to deal with different node types. 
(see: `library_analyzer/utils/_ast_walker.py`)
Unfortunately, this results in redundant code.

File: src/library_analyzer/processing/api/_resolve_references.py (hunk 3 above)
Comment (lukarade) [reply src/library_analyzer/processing/api/_resolve_references.py:154]: The separate functions are absolutely necessary. 
The AST is traversed by the ASTWalker, which needs a separate enter and leave function to deal with different node types. 
(see: `library_analyzer/utils/_ast_walker.py`)
Unfortunately, this results in redundant code.

**Review by nvollroth — APPROVED** [review]
Looks good to me.

**Pull request merged**